src/gui/widgets/VectorGraphViewBase.cpp

```diff
@@ -0,0 +1,204 @@
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include "VectorGraphViewBase.h"
+
+#include <vector>
+#include <QInputDialog> // showInputDialog()
+#include <QMenu> // context menu
+
+
+#include "StringPairDrag.h"
+#include "CaptionMenu.h" // context menu
+#include "embed.h" // context menu
+#include "MainWindow.h" // getting main window for context menu
+#include "GuiApplication.h" // getGUI
+#include "SimpleTextFloat.h"
+#include "AutomatableModel.h"
+#include "ControllerConnectionDialog.h"
+#include "ControllerConnection.h"
+
+
+namespace lmms
+{
+
+namespace gui
+{
+VectorGraphViewBase::VectorGraphViewBase(QWidget* parent) :
+	QWidget(parent),
+	m_hintText(new SimpleTextFloat)
+{
+}
+VectorGraphViewBase::~VectorGraphViewBase()
+{
+	delete m_hintText;
+}
+
+void VectorGraphViewBase::showHintText(QWidget* thisWidget, QString hintText, int msecBeforeDisplay, int msecDisplayTime)
+{
+	m_hintText->setText(hintText);
+	m_hintText->moveGlobal(thisWidget, QPoint(0, 0));
+	m_hintText->showWithDelay(msecBeforeDisplay, msecDisplayTime);
+}
+void VectorGraphViewBase::hideHintText()
+{
+	m_hintText->hide();
+}
+
+void VectorGraphViewBase::connectToAutomationTrack(QMouseEvent* me, FloatModel* automationModel, QWidget* thisWidget)
+{
+	if (automationModel != nullptr)
+	{
+		qDebug("mousePress automation sent");
+		new gui::StringPairDrag("automatable_model", QString::number(automationModel->id()), QPixmap(), thisWidget);
+		me->accept();
+	}
+}
+
+void VectorGraphViewBase::showContextMenu(const QPoint point, FloatModel* automationModel, QString displayName, QString controlName)
+{
+	m_curAutomationModel = automationModel;
+	CaptionMenu contextMenu(displayName + QString(" - ") + controlName);
+	addDefaultActions(&contextMenu, controlName);
+	contextMenu.exec(point);
+}
+void VectorGraphViewBase::addDefaultActions(QMenu* menu, QString controlDisplayText)
+{
+	// context menu settings
+	menu->addAction(embed::getIconPixmap("reload"),
+		tr("name: ") + controlDisplayText,
+		this, SLOT(contextMenuRemoveAutomation()));
+	menu->addAction(embed::getIconPixmap("reload"),
+		tr("remove automation"),
+		this, SLOT(contextMenuRemoveAutomation()));
+	menu->addSeparator();
+
+	QString controllerTxt;
+
+	menu->addAction(embed::getIconPixmap("controller"),
+		tr("Connect to controller..."),
+		this, SLOT(contextMenuExecConnectionDialog()));
+	if(m_curAutomationModel != nullptr && m_curAutomationModel->controllerConnection() != nullptr)
```

Comment:
The other codefactor checks seems to be false positives but the one below seems to be an actual fail.